src/containers/Rewards.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import {connect} from 'react-redux'
+import AddRewardForm from '../components/AddRewardForm';
+import {addContractData } from '../redux/actions/addReward';
+
+class AddReward extends Component{
+
+    state = {
+        amount: 0,
+        withdrawable: 0,
+    };
+
+    async componentDidMount(){
+        const {addContractDataFunc, contract: {data}} = this.props;
+        if (!data){
+            await addContractDataFunc()
+        }
+    };
+
+    handleChange = (e) => {
+        const {name, value} = e.target;
+        this.setState({[name]: value});
+    } 
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        const { amount, withdrawable} = this.state;
+        const {contract:{ data:{ owner, festaking}}} = this.props;
+
+        const totalReward =  await festaking.methods.totalReward().call();
+        console.log('total reward', totalReward);
+
+        const addReward = await festaking.methods.addReward(amount,withdrawable)
+        .send({from:owner, gas:500000});
+        
+
+        if(addReward){
+            const totalReward2 =  await festaking.methods.totalReward().call();
```

Comment:
1. Same here, use the contract proxy I noted above.

2. Also note that this method will **not** reflect the added reward until the previous transaction is confirmed. So you need to wait until transaction is confirmed.

It is better to design the UI with this in mind.